src/deploy/put.rs

```diff
@@ -33,6 +34,7 @@ impl ClientCommand for PutDeploy {
         let verbosity_level = common::verbose::get(matches);
 
         let secret_key = common::secret_key::get(matches);
+        let maybe_block_identifier = creation_common::speculative_exec::get(matches);
```

Comment:
With the other changes suggested, this will become `let (speculative_exec, maybe_block_identifier) =` and below we'll call `casper_client::cli::speculative_put_deploy` or `casper_client::cli::put_deploy` depending upon the bool.

Reply:
I took a slightly different but I hope, equivalent path, using an option type to encode the speculative_exec/no_speculative_exec distinction.